cmd/installer/pkg/install/manifest.go

```diff
@@ -57,11 +58,23 @@ type Target struct {
 	FileSystemType     FileSystemType
 	LegacyBIOSBootable bool
 
-	Size             uint
-	Force            bool
-	Assets           []*Asset
+	Size   uint
+	Force  bool
+	Assets []*Asset
+
+	// Preserve contents of the partition with the same label (if it exists).
 	PreserveContents bool
 
+	// Extra preserved locations (for upgrading from older versions of Talos).
+	//
+	// Used only if PreserveContents is true.
+	ExtraPreserveSources []PreserveSource
```

Comment:
this allows to copy `/boot/config.yaml` from 0.6 over to `/system/state/config.yaml` in 0.7